builder/cloudstack/ssh.go

```diff
@@ -7,28 +7,30 @@ import (
 
 	packerssh "github.com/hashicorp/packer/communicator/ssh"
 	"github.com/mitchellh/multistep"
-	"github.com/xanzy/go-cloudstack/cloudstack"
 	"golang.org/x/crypto/ssh"
 	"golang.org/x/crypto/ssh/agent"
 )
 
 func commHost(state multistep.StateBag) (string, error) {
-	client := state.Get("client").(*cloudstack.CloudStackClient)
-	config := state.Get("config").(*Config)
 
-	if config.hostAddress == "" {
-		ipAddr, _, err := client.Address.GetPublicIpAddressByID(config.PublicIPAddress)
```

Comment:
I looked it over and this logic is really still needed. If you supply the UUID of a public IP address instead of the IP address itself, `config.hostAddress` will never get the IP address and will remain empty.

You did indeed find and replace all places where `config.hostAddress` is given a value, but if you supplied a UUID, it will never receive a value and stay empty. So I guess we should make this something like:

```
func commHost(state multistep.StateBag) (string, error) {
	ipAddr, ok := state.Get("ipaddress").(string)
	if !ok {
		if !isUUID(config.PublicIPAddress) {
			return "", fmt.Errorf("Failed to retrieve IP address")
		}
		
		ip, _, err := client.Address.GetPublicIpAddressByID(config.PublicIPAddress)
		if err != nil {
			return "", fmt.Errorf("Failed to retrieve IP address: %s", err)
		}
		ipAddr = ip.Ipaddress
	}

	return ipAddr, nil
}
```